src/braket/experimental/autoqasm/instructions/gates.py

```diff
@@ -52,14 +55,14 @@ def cnot(
 def cphaseshift(
     control: QubitIdentifierType,
     target: QubitIdentifierType,
-    angle: float,
+    angle: Union[float, FreeParameterExpression],
```

Comment:
I wonder if there is a good way to define an alias for this union type. Seems like user-defined gates should also technically be using this as a type hint, and we wouldn't want users to have to copy this. Does it make sense to have an `AngleIdentifierType` that we expose as `aq.Angle` (just like `aq.Qubit` for `QubitIdentifierType`)? Or is that too specific - are there other types that should be supported with this pattern too? Gates can only take angles, but in general `input` parameters to a program can be other types.

Reply:
Ah I'm noticing now that `gate_calibrations` produce signatures with `angle`, so we should prioritize consistency using `angles` everywhere. I didn't realize they are already used in some places

Reply:
> Aaron pointed out that we'll want to migrate from float to OpenQASM angles, because gates are not supposed to take floats.

This is just a consideration for the IR we generate. Users will still certainly provide floats

Reply:
Yeah, gate definitions in OQ3 must use `angle` params, so we do use that for `@aq.gate` and `@aq.gate_calibration`. OQ3 allows implicitly casting `float` to `angle`, so technically everything else could use `float` and be fine, I think. But I also agree that if possible we should try to avoid exposing that to users, who just want to use `float` everywhere.

Reply:
Ah okay, wasn't aware of the implicit cast. That's good news